### PR TITLE
Hack#29: Show user's name in a Webflow Form success message

### PR DESCRIPTION
**Pull request by brianndiiri**

### CODE REVIEW

File: webflow-hacks/form-hacks/29/29.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// when DOM is ready
+document.addEventListener('DOMContentLoaded', function () {
+  // declare constant selectors
+  const FORM_SELECTOR = '[fs-hacks-element="hack29-form"]';
+  const NAME_INPUT_SELECTOR = '[fs-hacks-element="hack29-name-input"]';
+  const MESSAGE_SELECTOR = '[fs-hacks-element="hack29-custom-message"]';
```
Comment (alexiglesias93) [inline webflow-hacks/form-hacks/29/29.ts]: I assume this is the native `Success Message` element inside the form.
No need to add an attribute to query it, you can just use the native `.w-form-done` CSS class.

File: webflow-hacks/form-hacks/29/29.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  const nameInput = document.querySelector<HTMLInputElement>(NAME_INPUT_SELECTOR);
```
Comment (alexiglesias93) [inline webflow-hacks/form-hacks/29/29.ts]: I would query this inside the form element's scope.

**Review by alexiglesias93 — CHANGES_REQUESTED** [review]

**Comment by brayanfinsweet** [comment]
BTW can the anonymous function on the listener be better written as a pure function?

File: webflow-hacks/form-hacks/29/29.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+// when DOM is ready
+document.addEventListener('DOMContentLoaded', function () {
+  // declare constant selectors
+  const FORM_SELECTOR = '[fs-hacks-element="hack29-form"]';
+  const NAME_INPUT_SELECTOR = '[fs-hacks-element="hack29-name-input"]';
+  const MESSAGE_SELECTOR = '.w-form-done';
+  const messageDiv = document.querySelector<HTMLDivElement>(MESSAGE_SELECTOR);
```
Comment (alexiglesias93) [inline webflow-hacks/form-hacks/29/29.ts]: The `messageDiv` should be also queried under the form scope, as ther emight be more than 1 form on the page.

**Review by alexiglesias93 — CHANGES_REQUESTED** [review]

File: webflow-hacks/form-hacks/29/29.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  const FORM_SELECTOR = '[fs-hacks-element="hack29-form"]';
+  const NAME_INPUT_SELECTOR = '[fs-hacks-element="hack29-name-input"]';
+  const MESSAGE_SELECTOR = '.w-form-done';
+  const form = document.querySelector<HTMLFormElement>(FORM_SELECTOR);
+  // early return
+  if (!form) return;
+  // when form is submitted
+  form.addEventListener('submit', function () {
+    const nameInput = this.querySelector<HTMLInputElement>(NAME_INPUT_SELECTOR);
+    const messageDiv = this.querySelector<HTMLDivElement>(MESSAGE_SELECTOR);
```
Comment (alexiglesias93) [inline webflow-hacks/form-hacks/29/29.ts]: There's no reason to have those selectors inside the event listener, I think it's better to put them outside, right after the early return when the form is null.

**Review by alexiglesias93 — CHANGES_REQUESTED** [review]

File: webflow-hacks/form-hacks/29/29.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  const NAME_INPUT_SELECTOR = '[fs-hacks-element="hack29-name-input"]';
+  const MESSAGE_SELECTOR = '.w-form-done';
+  const form = document.querySelector<HTMLFormElement>(FORM_SELECTOR);
+  // early return
+  if (!form) return;
+  const nameInput = form.querySelector<HTMLInputElement>(NAME_INPUT_SELECTOR);
+  const messageDiv = form.querySelector<HTMLDivElement>(MESSAGE_SELECTOR);
+  // when form is submitted
+  form.addEventListener('submit', function () {
+    if (!nameInput || !messageDiv) return;
```
Comment (alexiglesias93) [inline webflow-hacks/form-hacks/29/29.ts]: Same here, why add the return inside the listener when you can directly stop the listener from ever being added?
You should put this return outside the listener

**Review by alexiglesias93 — CHANGES_REQUESTED** [review]

File: webflow-hacks/form-hacks/29/29.ts (hunk 5 above)
Comment (brianndiiri) [reply webflow-hacks/form-hacks/29/29.ts]: done

**Review by alexiglesias93 — APPROVED** [review]

**Pull request merged**